.github/workflows/build.yml

```diff
@@ -0,0 +1,159 @@
+        $DOCKER_COMPOSE exec -T web /bin/sh -c "bundle exec rake cc:setup"
+      env:
+        GIT_BRANCH: ${{env.BRANCH_TAG}}
+
+    - name: Run tests
+      run: |
+        $DOCKER_COMPOSE exec -T web /bin/sh -c 'bundle config --local disable_exec_load true'
+        $DOCKER_COMPOSE exec -T web /bin/sh -c 'bundle exec rake parallel:setup'
+        $DOCKER_COMPOSE exec -T web /bin/sh -c 'bundle exec rake "parallel:spec[,, -O .azure_parallel]"'
+      env:
+        GIT_BRANCH: ${{env.BRANCH_TAG}}
+
+    - name: Tag docker images
+      run: |
+        docker tag $DOCKER_IMAGE:$BRANCH_TAG $DOCKER_IMAGE:$SHA_TAG
+
+    - name: Push docker images to Docker Hub
+      run: |
+        docker push $DOCKER_IMAGE:$SHA_TAG
+        docker push $DOCKER_IMAGE:$BRANCH_TAG
+        [[ "$BRANCH_TAG" == "master" ]] && docker push $DOCKER_IMAGE-middleman:$BRANCH_TAG
+
+    - name: Install Terraform CloudFoundry Provider
+      run: |
+          mkdir -p $HOME/.terraform.d/plugins/linux_amd64
+          wget -O ${{ env.CF_PROVIDER }} https://github.com/cloudfoundry-community/terraform-provider-cf/releases/latest/download/terraform-provider-cloudfoundry_linux_amd64
+          chmod +x ${{ env.CF_PROVIDER }}
+      env:
+        CF_PROVIDER: $HOME/.terraform.d/plugins/linux_amd64/terraform-provider-cloudfoundry
+
+    - name: Terraform init
+      run: |
+          terraform init \
+           -backend-config=storage_account_name=${{ env.BACKEND_STORAGE_ACCOUNT_NAME }} \
+           -backend-config=access_key=${{ env.BACKEND_ACCESS_KEY }} \
+           -backend-config=key=${{ env.BACKEND_KEY }} \
+            ${{ env.PAAS_TF_DIR }}
+      env:
+        BACKEND_STORAGE_ACCOUNT_NAME: s121d01tfstatestr
+        BACKEND_ACCESS_KEY: ${{ secrets.BACKEND_ACCESS_KEY }}
+        BACKEND_KEY: paas-ttapi.tfstate
+        PAAS_TF_DIR: $HOME/work/teacher-training-api/teacher-training-api/terraform/paas
+
+    - name: Terraform plan
+      run: |
+          terraform plan -auto-approve \
+            -var-file=${{ env.PAAS_TF_DIR }}/${{ env.TERRAFORM_VAR_FILE }} \
+            ${{ env.PAAS_TF_DIR }}
+      env:
+        BACKEND_STORAGE_ACCOUNT_NAME: s121d01tfstatestr
+        BACKEND_ACCESS_KEY: ${{ secrets.BACKEND_ACCESS_KEY }}
+        BACKEND_KEY: paas-ttapi.tfstate
+        PAAS_TF_DIR: $HOME/work/teacher-training-api/teacher-training-api/terraform/paas
+        TERRAFORM_VAR_FILE: terraform_qa.tfvars
+        TF_VAR_ttapi_image_tag: ${{env.SHA_TAG}}
+        TF_VAR_api_url: https://api.london.cloud.service.gov.uk
+        TF_VAR_docker_image: ${{ env.DOCKER_IMAGE }}:${{ env.SHA_TAG }}
+        TF_VAR_user: ${{ secrets.TF_VAR_user }}
+        TF_VAR_password: ${{ secrets.TF_VAR_password }}
+        TF_VAR_SECRET_KEY_BASE: ${{ secrets.TF_VAR_SECRET_KEY_BASE }}
+        TF_VAR_SENTRY_DSN: ${{ secrets.TF_VAR_SENTRY_DSN }}
+        # Let's confirm if these variables are required
+        #TF_VAR_SETTINGS__GOOGLE__GCP_API_KEY: ${{ secrets.TF_VAR_SETTINGS__GOOGLE__GCP_API_KEY }}
```

Comment:
I just had a further thought about this one ... this key is used by the geocoding bg process, I would bet. That would probably mean we'll need it once we want to get that background job working.